ooni/agent/scheduler.py

```diff
@@ -0,0 +1,409 @@
+import os
+import errno
+
+from hashlib import md5
+from datetime import datetime
+
+from twisted.application import service
+from twisted.internet import defer
+from twisted.internet.task import LoopingCall
+from twisted.python.filepath import FilePath
+
+from ooni.scripts import oonireport
+from ooni import resources
+from ooni.utils import log, SHORT_DATE
+from ooni.utils.files import human_size_to_bytes, directory_usage
+from ooni.deck.store import input_store, deck_store, DEFAULT_DECKS
+from ooni.settings import config
+from ooni.contrib import croniter
+from ooni.contrib.dateutil.tz import tz
+
+from ooni.geoip import probe_ip
+from ooni.measurements import list_measurements
+
+class FileSystemlockAndMutex(object):
+    """
+    This is a lock that is both a mutex lock and also on filesystem.
+    When you acquire it, it will first block on the mutex lock and then
+    once that is released it will attempt to acquire the lock on the
+    filesystem.
```

Comment:
I am not sure the sentence above is correct. AFAICT the code does the following:

```
lock_mutex()
lock_fs()
do_something()
unlock_fs()
unlock_mutex()
```

but the sentence says "once [the mutex] is released it will attempt to acquire the lock on the fileystem. I think it should say instead, "once [the mutex] is acquired". Correct?